aardvark-doc/src/document.rs

```diff
@@ -92,9 +91,11 @@ mod imp {
 
         fn constructed(&self) {
             let service = self.service.get().unwrap();
-            let (network_tx, mut rx) = service.document(Hash::new(b"some id"));
+            let (document_id, network_tx, mut rx) = service.create_document();
             let public_key = service.public_key();
 
+            self.id.get_or_init(|| document_id.to_hex());
```

Comment:
This doesn't feel right, but I'm not sure what the right pattern is for GTK:

We don't want to pass in the document id on `Document::new` but get it here and then set the property inside the constructor.

@jsparber

Reply:
Maybe shouldn't have force pushed. Your code was mostly correct. I now added that the application can pass in an id. When an id is provided the document will be joined if `none` is provided a new document is created.